### PR TITLE
Feat pull function

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

Pulls existing functions from remote Appwrite server, both code and definition. 

## Test Plan

Against Appwrite version `1.5.5` 

https://github.com/appwrite/sdk-generator/assets/316103/82901a3c-1660-41a9-a956-35b13e3b3080

### CODE REVIEW

File: templates/cli/base/requests/api.twig [hunk 1]
```diff
@@ -10,12 +10,16 @@
     }, payload{% if method.type == 'location' %}, 'arraybuffer'{% endif %});
 
 {% if method.type == 'location' %}
```
Comment (loks0n) [inline templates/cli/base/requests/api.twig:12]: Just FYI, the whitespace slurping twig tags much look better here:
```twig
    {%~ if method.type == 'location' %}
    if (overrideForCli) {
        response = Buffer.from(response);
    }

    fs.writeFileSync(destination, response);
   {%~ endif %}
```

I like to fix them as I touch places in SDK generator

File: templates/cli/lib/commands/command.js.twig [hunk 2]
```diff
@@ -58,7 +59,7 @@ const {{ service.name | caseLower }} = new Command("{{ service.name | caseLower
 /**
  * @param {{ "{" }}{{ service.name | caseUcfirst }}{{ method.name | caseUcfirst }}RequestParams{{ "}" }} params
  */
-const {{ service.name | caseLower }}{{ method.name | caseUcfirst }} = async ({ {% for parameter in method.parameters.all %}{{ parameter.name | caseCamel | escapeKeyword }}, {% endfor %}parseOutput = true, sdk = undefined{% if 'multipart/form-data' in method.consumes %}, onProgress = () => {}{% endif %}{% if method.type == 'location' %}, destination{% endif %}}) => {
+const {{ service.name | caseLower }}{{ method.name | caseUcfirst }} = async ({ {% for parameter in method.parameters.all %}{{ parameter.name | caseCamel | escapeKeyword }}, {% endfor %}parseOutput = true, overrideForCli = false, sdk = undefined{% if 'multipart/form-data' in method.consumes %}, onProgress = () => {}{% endif %}{% if method.type == 'location' %}, destination{% endif %}}) => {
```
Comment (loks0n) [inline templates/cli/lib/commands/command.js.twig:62]: Is it possible to break this into multiple lines for each param?

File: templates/cli/lib/commands/pull.js.twig [hunk 3]
```diff
@@ -27,6 +30,64 @@ const pullProject = async () => {
...
+            localConfig.addFunction(func);
+        }
+
+        const localFunction = localFunctions.find((localFunc) => localFunc['$id'] === func['$id']);
+
+        if (localFunction['deployment'] === '') {
+            continue
+        }
+
+        const compressedFileName = `${+new Date()}.tar.gz`
```
Comment (loks0n) [inline templates/cli/lib/commands/pull.js.twig:69]: Should this be a parameter? Maybe it doesn't matter

Comment (byawitz) [reply templates/cli/lib/commands/pull.js.twig:70]: This is just a temp name, and the file get deleted after it is extracted.

File: templates/cli/lib/questions.js.twig [hunk 4]
```diff
@@ -311,6 +311,30 @@ const questionsPullProject = [
...
+                    name: `${func.name} (${func.$id})`,
+                    value: func
+                }
+            });
+        }
+    },
+    {
+        type: "input",
+        name: "override",
+        message: `Do you want to override local functions code and definition? ${chalk.red('all local changes will lost!')} Type "YES" to confirm.`
```
Comment (christyjacob4) [inline templates/cli/lib/questions.js.twig:334]:
```suggestion
        message: `Are you sure you want to override local functions code and definition? ${chalk.red('All local changes will be lost!')} Type "YES" to confirm.`
```

**Review by loks0n — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**